### DISCUSSION TITLE
Separating development for different versions of Jakarta EE

### DISCUSSION THREAD

**Discussion by zUniQueX** [qos-ch/logback#611] [Q&A] [answered]
As proposed in #585, Logback now separated the development of Jakarta EE 8 (Logback 1.3.x) and Jakarta EE 9+ (Logback 1.4.x).

In the `pom.xml` of the current `master`, there are some components of Jakarta EE 9 (like Servlet 5) and some components of Jakarta EE 10 (like Mail 2.1).

Are there any plans for further separation of Jakarta EE 9 and Jakarta EE 10 components?

**Comment by ceki** [comment, marked as answer]
@zUniQueX  There are no such plans.

If I remember correctly, Servlet 5 was used because of a dependency on either Jetty or Tomcat further down and Mail 2.1 was used because it was modularized.  

If the current component versions cause problems, please let me know.

**Reply by zUniQueX** [reply]
Thanks for the fast answer!

I'm just upgrading a library to Jakarta EE components and encountered a problem with Jetty because there is no version for Servlet 6 yet. To avoid conflicts, I've now decided to stay on one specific EE version and wanted to know, if Logback would probably provide seperate releases for different EE versions too.

Currently, I don't see problems with Logback. But if Jetty 12 will be released and Logback would directly upgrade to Servlet 6, there would be no compatible maintained version for Jetty 11. This may be undesirable.

**Reply by ceki** [reply]
I am quite tempted to _eventually_ move to Jetty 12 given that it uses Servlet 6 which is modularized.

**Reply by zUniQueX** [reply]
This would be a step in the right direction. If I remember correctly, there was a discussion on Jira supporting different versions of the webservers in different modules. Maybe this would be the way to go then.

**Reply by ceki** [reply]
Jetty 12 is still alpha though.

**Reply by zUniQueX** [reply]
Therefore I was asking now to avoid a breaking change when Jetty 12 will be released. Working around the `JettyModernServerAdapter` is generally possible (I'm using a workaround for Jetty 10 for now). But it would be nice, if Logback would provide continuous support for Jetty 11.
